crates/resolver/src/hosts.rs

```diff
@@ -47,14 +49,62 @@ impl Hosts {
     /// Look up the addresses for the given host from the system hosts file.
     pub fn lookup_static_host(&self, query: &Query) -> Option<Lookup> {
         if !self.by_name.is_empty() {
-            if let Some(val) = self.by_name.get(query.name()) {
-                let result = match query.query_type() {
-                    RecordType::A => val.a.clone(),
-                    RecordType::AAAA => val.aaaa.clone(),
-                    _ => None,
-                };
-
-                return result;
+            match query.query_type() {
+                RecordType::A | RecordType::AAAA => {
+                    if let Some(val) = self.by_name.get(query.name()) {
+                        let result = match query.query_type() {
+                            RecordType::A => val.a.clone(),
+                            RecordType::AAAA => val.aaaa.clone(),
+                            _ => None,
+                        };
+
+                        return result;
+                    }
+                }
+                RecordType::PTR => {
+                    if let Ok(ip) = query.name().parse_arpa_name() {
```

Comment:
Let's use `let-else` here to reduce rightward drift here, and also for the `hostnames.is_empty()` case.